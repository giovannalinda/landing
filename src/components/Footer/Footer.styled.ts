import styled from 'styled-components'
import { theme } from '~/styles'

export const Footer = styled.footer`
  width: 100%;
  margin: 16rem auto 17.8rem;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  small {
    margin-top: 1.8rem;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.gray500};

    a {
      color: ${theme.colors.secondaryPurple};
    }
  }
`

export const Separator = styled.hr`
  border: 0.1px solid ${theme.colors.gray100};
  width: 100%;
  opacity: 0.1;
  margin-top: 10.8rem;
  margin-bottom: 5.3rem;
`
