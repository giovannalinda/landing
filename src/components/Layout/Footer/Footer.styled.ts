import styled from 'styled-components'
import { theme } from '~/styles'

export const Footer = styled.footer`
  width: 100%;
  display: grid;
  place-items: center;
  padding-bottom: 17.8rem;

  small {
    margin-top: ${theme.space[4]};
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.gray500};

    a {
      color: ${theme.colors.secondaryPurple};
    }
  }
`

export const Separator = styled.hr`
  border: 0.1rem solid ${theme.colors.gray100};
  width: 100%;
  opacity: 0.1;
  margin-top: 10.8rem;
  margin-bottom: 5.3rem;
`
