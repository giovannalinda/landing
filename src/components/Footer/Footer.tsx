import Image from 'next/image'
import { Assets } from '~/assets'

import * as S from './Footer.styled'

export function Footer() {
  return (
    <S.Footer>
      <S.Separator />
      <Image src={Assets.Logo} width='263' height='59' />
      <small>
        Made with ðŸ’œ by John and{' '}
        <a href='https://www.hypesoft.com.br/'>Hypesoft</a>
      </small>
    </S.Footer>
  )
}
