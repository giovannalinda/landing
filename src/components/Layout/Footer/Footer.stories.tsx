import { Meta, Story } from '@storybook/react'

import { Footer } from './Footer'

export default {
  title: 'Components/Layout/Footer',
  component: Footer,
} as Meta

export const Default: Story = () => <Footer />
