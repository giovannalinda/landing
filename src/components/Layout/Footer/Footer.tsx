import Image from 'next/image'
import Link from 'next/link'
import { Assets } from '~/assets'

import * as S from './Footer.styled'

export function Footer() {
  return (
    <S.Footer>
      <S.Separator />
      <Image src={Assets.Logo} width='263' height='59' />
      <small>
        {'Made with ðŸ’œ by John and '}
        <Link href='https://www.hypesoft.com.br/' passHref>
          <a target='_blank' rel='noreferrer'>
            Hypesoft
          </a>
        </Link>
      </small>
    </S.Footer>
  )
}
