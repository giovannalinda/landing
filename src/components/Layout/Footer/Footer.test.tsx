import { render } from '@testing-library/react'
import { Footer } from './Footer'

describe('Footer', () => {
  it('should match snapshot', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
