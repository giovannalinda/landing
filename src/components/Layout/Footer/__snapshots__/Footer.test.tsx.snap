// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<footer
  class="sc-bczRLJ fsmWSS"
>
  <hr
    class="sc-gsnTZi ixoeZt"
  />
  <img
    height="59"
    src="/logo.svg"
    width="263"
  />
  <small>
    Made with
     ðŸ’œ 
    by Giovanna and 
    <a
      href="https://www.hypesoft.com.br/"
      rel="noreferrer"
      target="_blank"
    >
      Hypesoft
    </a>
  </small>
</footer>
`;
